### PR TITLE
Re-implement `createEntity` BP hook

### PR DESCRIPTION
**Pull request by benwerner01**
## 🌟 What is the purpose of this PR?

This PR re-implements the BP `createEntity` hook using the new graph service.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202953520344821/1203109184120791/f) _(internal)_

## 🔍 What does this change?

See commits.

## 📜 Does this require a change to the docs?

The BP method documentation will need to be updated for the new type system once their definitions are stable.

## 🛡 What tests cover this?

None.

## ❓ How to test this?

1. Checkout the branch
2. Run the external services and `yarn dev`
3. Go to `localhost:3000` and log in
4. Go to `localhost:3000/entity-editor`
5. Click the "Create Entity" button (which uses the BP hook), and check the console to inspect the created entity

## 📹 Demo

https://user-images.githubusercontent.com/42802102/194828638-6b08d217-4307-4d23-8486-c16601ca2e59.mov

### CODE REVIEW

File: packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolCreateEntity.ts [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+      const { data: responseData } = await createFn({
+        variables: {
+          entityTypeId,
+          properties,
+        },
+      });
+
+      const { createPersistedEntity: createdEntity } = responseData ?? {
+        createPersistedEntity: null,
```
Comment (thehabbos007) [inline packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolCreateEntity.ts:63]: I suppose this suggestion is similar to what we did for `updateEntity`, could we have empty braces here instead? 
like in https://github.com/hashintel/hash/blob/4f14a1ec27bade45b62994508d86ab71bdb32ef8/packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolUpdateEntity.ts#L61

File: packages/hash/frontend/src/pages/entity-editor.page.tsx [hunk 2]
```diff
@@ -62,9 +63,23 @@ const ExampleUsage = ({ ownedById }: { ownedById: string }) => {
...
+    const createdEntity = await createEntity({
+      data: {
+        entityTypeId:
+          "http://localhost:3000/@example/types/entity-type/dummy/v/1",
+        properties: {},
+      },
+    });
+
+    // eslint-disable-next-line no-console
+    console.log({ createdEntity });
```
Comment (thehabbos007) [inline packages/hash/frontend/src/pages/entity-editor.page.tsx:76]: Is this `console.log` intentional for demo purposes?

**Review by thehabbos007 — COMMENTED** [review]
Looks good to me, just have a couple comments, I'd be happy to approve after these are addressed :)

File: packages/hash/frontend/src/pages/entity-editor.page.tsx (hunk 2 above)
Comment (benwerner01) [reply packages/hash/frontend/src/pages/entity-editor.page.tsx:76]: Yes, it's so there's a way of testing this PRs functionality (see the GitHub description). This demo page should be removed when @yusufkinatas's entity editor is merged.

File: packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolCreateEntity.ts (hunk 1 above)
Comment (benwerner01) [reply packages/hash/frontend/src/components/hooks/blockProtocolFunctions/knowledge/useBlockProtocolCreateEntity.ts]: Thanks for spotting, fixed in https://github.com/hashintel/hash/pull/1178/commits/cf32e2e80595952918fd0e4d1609549d30d8f8da

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**